### PR TITLE
[GSOC-87] Robot models with sensor plugins

### PR DESCRIPTION
**Pull request by rakeshv24**
This PR includes the following updates:

1. Added Gazebo plugins for the sensor and DVL systems to the world files.
2. Removed the redundant DVL world file.
3. Made a minor modification to the topic name in the sea-pressure sensor plugin's source code.
4. Integrated all relevant sensor plugins for the `rexrov` and `glider_slocum` models.
5. Created separate launch files for each robot model to initialize a `ros_gz_bridge` node properly.
6. Combined `dave_robot_launch` package with `dave_demos` for generality. 

Documentation: [Link](https://www.notion.so/yeongdocat/Demo-Launch-Files-ebbd5c11ee08446f970df9f9f8fc8deb?pvs=4)

### CODE REVIEW

File: models/dave_robot_models/launch/upload_robot.launch.py [hunk 1]
```diff
@@ -128,10 +138,36 @@ def generate_launch_description():
...
+                    "launch",
+                    "glider_slocum.launch.py",
+                ]
+            )
+        ),
+        condition=IfCondition(EqualsSubstitution(namespace, "glider_slocum")),
+    )
+
+    include = [rexrov_launch, glider_launch]
+
```
Comment (woensug-choi) [inline models/dave_robot_models/launch/upload_robot.launch.py]: @rakeshv24 Could we have those `ros_gz_bridge` launch files in the each robot's description folder along with the `model.sdf`? I think those ros_gz configurations are more like descriptions for each robot

Example structure:

- dave_robot_models
  - description
    - glider_slocum
       - model.sdf
       - `ros_gz_bridge.py`
    - rexrov
       - model.sdf
       - `ros_gz_bridge.py`

**Review by woensug-choi — CHANGES_REQUESTED** [review]

File: models/dave_robot_models/launch/upload_robot.launch.py (hunk 1 above)
Comment (woensug-choi) [reply models/dave_robot_models/launch/upload_robot.launch.py]: This way, we don't need `IfCondition` and keep `upload_robot.launch.py` unchanged when adding new robots

Comment (rakeshv24) [reply models/dave_robot_models/launch/upload_robot.launch.py]: Sure that is a good idea. Let me try that out!

Comment (rakeshv24) [reply models/dave_robot_models/launch/upload_robot.launch.py]: @woensug-choi but how do we invoke those `ros_gz_bridge.py` files? Don't we still need to use the `IfCondition` for it?

Comment (woensug-choi) [reply models/dave_robot_models/launch/upload_robot.launch.py]: @rakeshv24 something like this?
```
    ros_gz_bridge = [IncludeLaunchDescription(
        PythonLaunchDescriptionSource(
            PathJoinSubstitution(
                [
                    FindPackageShare("dave_robot_models"),
                    "description",
                    namespace,
                    "ros_gz_bridge.py",
                ]
            )
        ),
    )]
```

Comment (rakeshv24) [reply models/dave_robot_models/launch/upload_robot.launch.py]: Sounds good!

**Pull request closed without merging**